src/main.rs

```diff
@@ -421,22 +437,35 @@ fn notmain() -> Result<i32, anyhow::Error> {
     let debug_frame = debug_frame.ok_or_else(|| anyhow!("`.debug_frame` section not found"))?;
 
     // print backtrace
-    let top_exception = backtrace(&mut core, pc, debug_frame, &range_names)?;
+    let top_exception = backtrace(
+        &mut core,
+        pc,
+        debug_frame,
+        &range_names,
+        &vector_table,
+        &sp_ram_region,
+    )?;
 
     core.reset_and_halt(TIMEOUT)?;
 
-    Ok(if top_exception == Some(TopException::HardFault) {
-        SIGABRT
-    } else {
-        0
-    })
+    Ok(
+        if let Some(TopException::HardFault { stack_overflow }) = top_exception {
+            if stack_overflow {
+                println!("{}", "!!!! STACK OVERFLOW DETECTED !!!!".red().bold());
```

Comment:
Hmm, this seems a bit over-the-top imo. Maybe something like this is better (somewhat closer to the libstd runtime message)?

```suggestion
                log::error!("the program has overflowed its stack");
```

Because of the `log::error`, we still get a red `ERROR` marker in the output.